alts/worker/runners/base.py

```diff
@@ -857,7 +857,21 @@ def detect_protected_packages(self):
             file_protected = [i.strip() for i in stdout.split('\n') if i.strip()]
             if file_protected:
                 protected.extend(file_protected)
-        return protected
+        protected.append('kernel-core')
+        dnf_command = (
+            r'dnf', '-q', '--qf=%{NAME}', 'repoquery', '--requires', '--resolve', '--recursive' +
+            ' '.join(protected)
+        )
+        exit_code, stdout, stderr = self.exec_command(*dnf_command)
+        if exit_code != 0:
+            self._logger.warning(
+                'Cannot resolve non-uninstallable packages via DNF: %s',
+                dnf_command
+            )
+        dnf_protected = [i.strip() for i in stdout.split('\n') if i.strip()]
```

Comment:
question. Let's say that, for whatever reason, this command execution fails. Shall we protect access to `exit_code` and `stdout` against this possibility?

Reply:
Good catch, added `return` when we failed the command